site/app/libraries/socket/Server.php

```diff
@@ -17,32 +17,37 @@ class Server implements MessageComponentInterface {
     // Holds the mapping between Connection Objects (key) and User_ID (value)
     private $sessions;
 
-    // Holds the mapping between Connection Objects IDs (key) and user current course (value)
-    private $courses;
+    // Holds the mapping between Connection Objects IDs (key) and user current course/page (value)
+    private $pages;
 
     // Holds the mapping between User_ID (key) and Connection object (value)
-    /** @var array<string, \Ratchet\ConnectionInterface> */
+    /** @var array<string, ConnectionInterface> */
     private $users;
 
     /** @var Core */
     private $core;
 
     public function __construct(Core $core) {
         $this->clients = [];
-        $this->courses = [];
+        $this->pages = [];
         $this->sessions = [];
-
+        $this->users = [];
         $this->core = $core;
     }
 
     /**
      * This function checks if a given connection object is authenticated
      * It uses the submitty_session cookie in the header data to work
+     * @param ConnectionInterface $conn
+     * @return bool
      */
     private function checkAuth(ConnectionInterface $conn): bool {
         $request = $conn->httpRequest;
-        $client_id = $conn->resourceId;
-        $origin = $request->getHeader('origin')[0];
+        $user_agent = $request->getHeader('User-Agent')[0];
+
+        if ($user_agent === 'websocket-client-php') {
+            return true;
+        }
```

Comment:
Are there any security implications of this?  Would it be possible for a bad actor to fake their user agent to be `websocket-client-php` and gain authorization to the websocket server?

Reply:
Yes. This has security implications. Using the above code, we should validate that if the user_agent is `websocket-client-php` and that `$request->getHeader('Session-Secret')[0] === $this->core->getConfig()->getSecretSession()`.